sklearn/neighbors/lof.py

```diff
@@ -0,0 +1,265 @@
+# -*- coding: utf-8 -*-
+
+import numpy as np
+
+from .base import NeighborsBase
+from .base import KNeighborsMixin
+from .base import UnsupervisedMixin
+
+from ..metrics import pairwise_distances
+
+from ..utils import check_array
+from ..utils import _get_n_jobs
+
+__all__ = ["LOF"]
+
+
+class LOFMixin(object):
```

Comment:
why do you need a mixin here?

I would put all methods from Mixin into LOF class and make them private.